test/Streamly/Test/Data/Stream/Concurrent.hs

```diff
@@ -236,50 +242,78 @@ main = hspec
                 (fmap (+1) . Async.eval . fmap (+1))
 
         asyncSpec $ prop "sequenceWith" . sequenceReplicate
-        -- XXX Need to use asyncSpec in all tests
-        prop "mapM (+1)" $
-            transform (fmap (+1)) (Async.mapM (\x -> return (x + 1)))
+        asyncSpec $ prop "mapM all configs mapM (+1)"
+                    . transform (fmap (+1))
+                    . (`Async.mapMWith` (\x -> return (x + 1)))
 
         -- XXX Need to use eq instead of sortEq for ahead oeprations
         -- Binary append
-        prop1 "append [] []"
-            $ cmp (Async.append [Stream.nil, Stream.nil]) sortEq []
-        prop1 "append [] [1]"
-            $ cmp (Async.append [Stream.nil, Stream.fromPure 1]) sortEq [1]
-        prop1 "append [1] []"
-            $ cmp (Async.append [Stream.fromPure 1, Stream.nil]) sortEq [1]
-        prop1 "append [0] [1]"
-            $ let stream = Async.append [Stream.fromPure 0, Stream.fromPure 1]
-               in cmp stream sortEq [0, 1]
-
-        prop1 "append [0] [] [1]"
-            $ let stream =
-                    Async.append
-                        [Stream.fromPure 0, Stream.nil, Stream.fromPure 1]
-               in cmp stream sortEq [0, 1]
-
-        prop1 "append2 left associated"
-            $ let stream =
-                    Stream.fromPure 0
-                        `Async.append2` Stream.fromPure 1
-                        `Async.append2` Stream.fromPure 2
-                        `Async.append2` Stream.fromPure 3
-               in cmp stream sortEq [0, 1, 2, 3]
-
-        prop1 "append right associated"
-            $ let stream =
-                    Stream.fromPure 0
-                        `Async.append2` (Stream.fromPure 1
-                        `Async.append2` (Stream.fromPure 2
-                        `Async.append2` Stream.fromPure 3))
-               in cmp stream sortEq [0, 1, 2, 3]
-
-        prop1 "append balanced"
-            $ let leaf x y = Stream.fromPure x `Async.append2` Stream.fromPure y
-                  leaf11 = leaf 0 1 `Async.append2` leaf 2 (3 :: Int)
-                  leaf12 = leaf 4 5 `Async.append2` leaf 6 7
-                  stream = leaf11 `Async.append2` leaf12
-               in cmp stream sortEq [0, 1, 2, 3, 4, 5, 6,7]
+        asyncSpec $
+            let appWith cfg = Async.combineWith cfg Stream.nil Stream.nil
+            in prop1 "mapM all configs append [] []"
+                    . cmp2 sortEq [] . appWith
+
+        asyncSpec $
+            let appWith cfg = Async.combineWith cfg Stream.nil (Stream.fromPure 1)
+            in prop1 "mapM all configs append [] [1]"
+                    . cmp2 sortEq [1] . appWith
+
+
+        asyncSpec $
+            let appWith cfg = Async.combineWith
+                        cfg Stream.nil (Stream.fromPure 1)
```

Comment:
Does this correspond to `cmp (Async.append [Stream.fromPure 1, Stream.nil]) sortEq [1]`?